src/drt/src/dr/FlexDR.cpp

```diff
@@ -104,6 +173,151 @@ int FlexDRWorker::main(frDesign* design)
+        return nullptr;
+      for (auto& blkg : inst->getInstBlockages())
+        if (blkg->getId() == srBlkg->getId())
+          return blkg.get();
+      return nullptr;
+    }
+    case frcBlockage: {
+      frBlockage* srBlkg = (static_cast<frBlockage*>(srObj));
+      for (auto& blkg : design->getTopBlock()->getBlockages())
+        if (blkg->getId() == srBlkg->getId())
+          return blkg.get();
+    }
+    default:
+      return nullptr;
+  }
+}
+void FlexDRWorker::updateDesign(frDesign* design)
+{
+  design_ = design;
+  for (auto& drNet_ : nets_) {
+    auto frNet_ = design->getTopBlock()->findNet(drNet_->getFrNet()->getName());
+    if (frNet_ != nullptr) {
+      if (drNet_->getFrNet()->isModified())
+        frNet_->setModified(true);
+      drNet_->setFrNet(frNet_);
+    } else {
+      logger_->error(DRT,
+                     501,
+                     "No matching net in design for net {}",
+                     drNet_->getFrNet()->getName());
+    }
+  }
+  map<frNet*, set<pair<Point, frLayerNum>>, frBlockObjectComp> bp;
+  for (auto [net, value] : boundaryPin_) {
+    auto frNet_ = design->getTopBlock()->findNet(net->getName());
+    if (frNet_ != nullptr) {
+      bp[frNet_] = value;
+    } else {
+      logger_->error(
+          DRT, 502, "No matching net in design for net {}", net->getName());
+    }
+  }
+  setBoundaryPins(bp);
+  std::vector<frMarker>& markers = getBestMarkers();
+  for (auto& marker : markers) {
+    std::set<frBlockObject*> newSrcs;
+    for (auto src : marker.getSrcs()) {
+      if (src != nullptr) {
+        auto newSrc = getEqObject(src, design);
+        if (newSrc == nullptr)
+          logger_->error(DRT, 503, "Fail in mapping serialized marker src");
+        newSrcs.insert(newSrc);
+      } else
+        newSrcs.insert(nullptr);
+    }
+    marker.setSrcs(newSrcs);
+    for (auto& [obj, value] : marker.getVictims()) {
+      if (obj != nullptr) {
+        auto newObj = getEqObject(obj, design);
+        if (newObj == nullptr)
+          logger_->error(DRT, 504, "Fail in mapping serialized marker victim");
+        obj = newObj;
+      }
+    }
+    for (auto& [obj, value] : marker.getAggressors()) {
+      if (obj != nullptr) {
+        auto newObj = getEqObject(obj, design);
+        if (newObj == nullptr)
+          logger_->error(
+              DRT, 505, "Fail in mapping serialized marker aggressor");
+        obj = newObj;
+      }
+    }
+  }
+}
+
+void FlexDRWorker::distributedMain(frDesign* design, const char* globals_path)
+{
+  ProfileTask profile("DR:main");
+  if (VERBOSE > 1) {
+    stringstream ss;
+    ss << endl
+       << "start DR worker (BOX) "
+       << "( " << routeBox_.xMin() * 1.0 / getTech()->getDBUPerUU() << " "
+       << routeBox_.yMin() * 1.0 / getTech()->getDBUPerUU() << " ) ( "
+       << routeBox_.xMax() * 1.0 / getTech()->getDBUPerUU() << " "
+       << routeBox_.yMax() * 1.0 / getTech()->getDBUPerUU() << " )" << endl;
+    cout << ss.str() << flush;
+  }
```

Comment:
This should happen via the logging library.

Reply:
Agreed, will update main(frDesign*) as well.